server/src/test/java/io/seata/server/store/db/DbcpDataSourceProviderTest.java

```diff
@@ -0,0 +1,37 @@
+/*
+ *  Copyright 1999-2019 Seata.io Group.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.server.store.db;
+
+import io.seata.common.loader.EnhancedServiceLoader;
+import io.seata.core.store.db.DataSourceProvider;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import javax.sql.DataSource;
+
+/**
+ * @author will
+ */
+public class DbcpDataSourceProviderTest {
+
+    private final String datasourceType = "dbcp";
+
+    @Test
+    public void testDbcpDataSourceProvider() {
```

Comment:
You can put the test case in the abstract class

Reply:
fixed.